python/tchannel/exceptions.py

```diff
@@ -54,6 +54,11 @@ class InvalidMessageException(ProtocolException):
     pass
 
 
+class InvalidEndpointException(ProtocolException):
+    """Represent an message containing invalid endpoint."""
+    pass
```

Comment:
From your original implementation, ProtocolException is only for error frame exceptions. ProtocolExceptions have a code, description, id, and tracing. 

Also, you're instantiating `InvalidEndpointException(..)` with just the message. The constructor for `ProtocolException` accepts at least 2 arguments. This won't work anyway. Most likely same with `InvalidMessageException`.